article/results.tex

```diff
@@ -56,59 +56,75 @@ \subsection{Best Practice Guidelines}
 As part of this work we have contributed substantial changes to the original OPFVTA repository, based on which we formulate a number of best practice guidelines, highly relevant in the production of reexecutable research outputs.
 
 \subsubsection{Errors should be fatal more often than not}
-The high complexity of a full analysis workflow makes missing output data intractable, as there are many steps at which data may have been dropped.
-Manually setting individual invocations propagate exit codes throughout the Make system is cumbersome, and thus an easy global invocation should be relied on so that errors become fatal and immediately visible.
-This is best accomplished by prepending all Bash scripts for which this concern is relevant with a \texttt{set -eu} line.
+
+By default any Python script or a compiled C or C++ program just errors out and exits immediately when running into some unexpected operation.
```

Comment:
Do we really want to be specific regarding the language? I assume it's the same for Rust and others, no? Maybe just “the majority of compiled and interpreted languages ..., though Shell in particular behaves differently.”

Reply:
please use suggestions... what about 
```suggestion
By default any Python script or a compiled, e.g. C or C++, program just errors out and exits immediately when running into some unexpected operation.
```
? as for having an example mentioning popular language -- could bring reader "home" and thus make it easier to relate.

Reply:
“The majority of programming languages (including e.g. Python and C) will by default exit immediately with an error when encountering unexpected operations.
Unix shell languages, by contrast, continue with execution of the next scripted command, and return non-0 exit codes only when the list of commands is exhausted or the `exit` command is called.”

Also I think it's important to make “shell” more specific. Any interpreted language can have a shell, and even some compiled languages (e.g. haskell) have shells. Also I am not sure whether it is all Unix shells which behave like this (POSIX, Bash, csh...) or just Bash, since I only ever use Bash.

Reply:
> The majority of programming languages (including e.g. Python and C) will by default exit immediately ...

languages do not exit IMHO since they do not run -- executable scripts or programs do.

re shell - good point. Please suggest (and I mean - use `suggestion` - 
![image](https://github.com/con/opfvta-replication-2023/assets/39889/30a4a3ba-424c-43c0-af0c-39ccd5926c4b)
) the changes next time... let me suggest one for shell below, feel welcome to suggest another improvement.